.github/workflows/spellcheck.yml

```diff
@@ -1,24 +1,37 @@
-name: Codespell
+name: 🚥 Codespell
+
+# Workflow to check spelling errors on the changed files within the current pull request.
+#
+# The previous version already did a complete check on the whole repository, therefore,
+# we prefer only to check the added/changed files for spelling errors.
+
 on:
-  push:  
   pull_request:
-    branches:
-      - master
-  workflow_dispatch: {}
+    types: [opened, edited, synchronize]
+
 jobs:
-  spellcheck:
-    name: Spell Check
-    runs-on: ubuntu-22.04
+  codespell:
+    runs-on: ubuntu-latest
     steps:
-    - name: Checkout
-      uses: actions/checkout@v4
-      with:
-        fetch-depth: 1
-    - name: Set up Python 🐍
-      uses: actions/setup-python@v5
-    - name: Install codespell
-      run: |
-        python -m pip install --upgrade pip
-        pip install codespell
-    - name: Check spelling
-      run: codespell --skip=".git,./renovate.json,./LICENSE,./MAINTAINERS,*.8,*.pdf,*.html,*.yml" --check-filenames --ignore-regex=.te# --ignore-words=.codespellignore
+      - uses: actions/checkout@v4
+
+      - name: 🔸 Get Changed Files from Pull Request
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          # get file names and seperate them with space ' '
+          files=$(gh pr diff ${{ github.event.pull_request.number }} --name-only | xargs -I {} sh -c 'echo "./{}"' | tr '\n' ' ')
+          echo "CHANGED_FILES=$files" >> "$GITHUB_ENV"          
+
+      - name: 🔸 Output Changed Files
+        run: echo ${{ env.CHANGED_FILES }}
+          
+      - name: 🔸 Check for Spelling Errors for Changed Files
+        uses: codespell-project/actions-codespell@v2
+        with:
+          check_filenames: true
+          check_hidden: true
+          ignore_words_file: .codespellignore
+          skip: nothing
+          path: ${{ env.CHANGED_FILES }}
```

Comment:
@gdha do you see any chance to harden that against paths/files with spaces or stupid chars in them?

Reply:
Let me check on this